test/jdk/java/nio/channels/FileChannel/CloseDuringTransfer.java

```diff
@@ -0,0 +1,418 @@
+            } catch (ClosedChannelException e) {
+                assertFalse(src.isOpen());
+            }
+            assertTrue(target.isOpen());
+        }
+    }
+
+    /**
+     * Close target channel during transferTo.
+     */
+    @ParameterizedTest
+    @MethodSource("targets")
+    void testCloseTargetDuringTransferTo(WritableByteChannel target) throws Exception {
+        try (FileChannel src = fileChannelSource(SOURCE_SIZE); target) {
+            scheduleClose(target);
+            try {
+                long n = src.transferTo(0, Long.MAX_VALUE, target);
+                assertTrue(n > 0);
+            } catch (ClosedChannelException e) {
+                assertFalse(target.isOpen());
+            }
+            assertTrue(src.isOpen());
+        }
+    }
+
+    /**
+     * Interrupt thread during transferTo.
+     */
+    @ParameterizedTest
+    @MethodSource("targets")
+    void testInterruptDuringTransferTo(WritableByteChannel target) throws Exception {
+        try (FileChannel src = fileChannelSource(SOURCE_SIZE); target) {
+            Future<?> interrupter = scheduleInterrupt();
+            try {
+                long n = src.transferTo(0, Long.MAX_VALUE, target);
+                assertTrue(n > 0);
+            } catch (ClosedByInterruptException e) {
+                assertTrue(Thread.currentThread().isInterrupted());
+                assertFalse(src.isOpen());
+                assertFalse(target.isOpen());
+            } finally {
+                finishInterrupt(interrupter);
+            }
+        }
+    }
+
+    /**
+     * Close source channel during transferFrom.
+     */
+    @ParameterizedTest
+    @MethodSource("sources")
+    void testCloseSourceDuringTransferFrom(ReadableByteChannel src) throws Exception {
+        try (src; FileChannel target = fileChannelTarget()) {
+            scheduleClose(src);
+            try {
+                long n = target.transferFrom(src, 0, Long.MAX_VALUE);
+                assertTrue(n > 0);
+            } catch (ClosedChannelException e) {
+                assertFalse(src.isOpen());
+            }
+            assertTrue(target.isOpen());
+        }
+    }
+
+    /**
+     * Close target file channel during transferFrom.
+     */
+    @ParameterizedTest
+    @MethodSource("sources")
+    void testCloseTargetDuringTransferFrom(ReadableByteChannel src) throws Exception {
+        try (src;  FileChannel target = fileChannelTarget()) {
+            scheduleClose(target);
+            try {
+                long n = target.transferFrom(src, 0, Long.MAX_VALUE);
+                assertTrue(n > 0);
+            } catch (ClosedChannelException e) {
+                assertFalse(target.isOpen());
+            }
+            assertTrue(src.isOpen());
+        }
+    }
+
+    /**
+     * Interrupt thread during transferFrom.
+     */
+    @ParameterizedTest
+    @MethodSource("sources")
+    void testInterruptTransferDuring(ReadableByteChannel src) throws Exception {
```

Comment:
I think this method must've been intended to be named `testInterruptDuringTransferFrom`.